### ISSUE TITLE
`z.record(z.any())` generates invalid OpenAPI spec

### ISSUE THREAD

**Issue by maxcohn** [RobinTail/express-zod-api#900]
Hi, I've run into some trouble with OpenAPI spec generation of input and output payloads. My goal is to accept an arbitrary object as input. The way I've found that I can accomplish this is with `z.record(z.any())` and that works just fine functionality wise. The issue I run into is that when I generate the docs, I get the following YAML in the OpenAPI spec that's generated:

```yaml
type: object
additionalProperties:
  format: any
  nullable: true
```

When I check the validity of this using a tool like Redocly's CLI, I get the following error:

```
The `type` field must be defined when the `nullable` field is used.

687 |   additionalProperties:
688 |     format: any
689 |     nullable: true
```

From looking at the OpenAPI spec, this is an accurate error. I was wondering if there was a known workaround/better way to handle a schema field for accepting an arbitrary object. Just using `z.any()` results in the same error as above.

If I manually edit the spec and add `type: object` to the `additionalProperties`, the spec stops throwing that error. Here's an example:

```
type: object
additionalProperties:
  format: any
  nullable: true
  type: object
```

I'm wondering if this is an issue with spec generation, or if I'm just trying to tackle this problem the wrong way. If you have any advice on how I should be using Zod to accept an arbitrary object that passes the OpenAPI spec, I'd love to hear.

Thank you!

**Comment by maxcohn** [comment]
I'm also curious if a possible solution would be to make the field `any` without allowing it to be nullable. Not sure if that's a valid option though.

**Comment by RobinTail** [comment]
Hello @maxcohn ,

could you tell me please, which version you're using?

**Comment by RobinTail** [comment]
> I'm also curious if a possible solution would be to make the field any without allowing it to be nullable. Not sure if that's a valid option though.

Well, `any` means literally any, implying the it can be `null` as well.

**Comment by RobinTail** [comment]
> When I check the validity of this using a tool like Redocly's CLI, I get the following error:
> ```The `type` field must be defined when the `nullable` field is used.```

Ok, that's interesting.
I've been using https://editor.swagger.io/ for validating against OpenAPI 3.0.
Maybe your tool using 3.1 ?
And also I'm using `openapi3-ts` lib for creating the specifications, which allows `type` to be omitted.
I'm going to check it.

> From looking at the OpenAPI spec, this is an accurate error.

Could you please send me a link to the place where you found this information?
@maxcohn

**Comment by maxcohn** [comment]
@RobinTail Thanks for getting back to me! I'm using version 8.8.1.

I should correct myself, that OpenAPI spec I was following was from Redocly specifically, which I'm assuming has an implementation difference from the swagger editor (which I assume is the more accurate to the spec since it's from Swagger). This sounds like it's more of an issue with Redocly at this point, but I'm not positive.

Apologies for taking up your time, but I appreciate your help!

**Comment by RobinTail** [comment]
It's OK , @maxcohn .
I'm actually curious how to specify `any` in a proper way, so it could have maximum compatibility.
I'm to try it without both `type` and `nullable`, to see if `null` would be accepted by Swagger UI.
I'll let you know on the results of this exaperiment.

**Comment by RobinTail** [comment]
Checked the details, it's really should not be `nullable` when no `type`

> A true value adds "null" to the allowed type specified by the type keyword, only if type is explicitly defined within the same Schema Object. Other Schema Object constraints retain their defined behavior, and therefore may disallow the use of null as a value. A false value leaves the specified or default type unmodified. The default value is false.

I'm going to fix it

**Comment by RobinTail** [comment]
🚀 The fix is about to be released in version 10.1.1, @maxcohn 
Thank you

**Comment by RobinTail** [comment]
Also fixed for v9 in v9.4.3 🚀

**Comment by maxcohn** [comment]
@RobinTail No, thank you for putting together this fantastic package!
